### PR TITLE
Update Windows build system to support MSYS2 builds properly

### PR DESCRIPTION
**Pull request by bagong**
Depends on three other PRs in SC (@awson #2473), HIDAPI ([#5](https://github.com/supercollider/hidapi/pull/5) @awson too), portaudio ([#1](https://github.com/supercollider/portaudio/pull/1) by me).

I've built this on all build variants for Windows and on MacOS Sierra and Ubuntu 16.10. No side effects detected. This is Windows only! Once the whole bundle of PRs is merged we'll first time have a (close to) feature-complete Win64 build including SuperNova. Also MinGW builds should work throughout the GCC-line from 4.8 to 6.3 (4.82, 4.92, 5.3 and 6.3 verified). Thanks again @awson!

If you want to test this, it's a bit involved as you need everything at the same time. I think the easiest way is like so (from memory, I hope no mistake):

in SC root-source folder:

````
git fetch origin pull/2704/head:prMSYSBuild
git checkout prMSYSBuild
cd external_libraries/hidapi
git fetch origin pull/5/head:pr530
git checkout pr530
cd ../portaudio_sc_org
git fetch origin pull/1/head:prPA
git checkout prPA
cd ../..
git fetch origin pull/2473/head:pr492
git merge pr492
````

ready to build, confirm with `git status`

I'd like to remark, that all of this is purely within the Windows domain, and build-stuff only that extends stuff I've contributed in the past without much outside involvement. So I doubt there will be much reviewing (I 'd be grateful for input, of course). Also: portaudio got a new release recently which made all of the pa build-hackery in Win3.7/3.8 unnecessary. We're building with vanilla PA now, with just a few changes to the top level CMakeLists.txt. Big thanks to the PA team too, the new release was a pleasure to unfold.

It would be nice if this was committed quickly, as it forms the base for ongoing SCWin3.9 initiatives and is quite complex to assemble for people who are not pretty git adept. The PRs mentioned depend on each other. I'll follow up problems, of course, but I think it's pretty well tested. Thanks!

### CODE REVIEW

**Comment by bagong** [comment]
Not sure why @snappizz ' remove Module is listed here, it's in master already...

**Comment by bagong** [comment]
Sorry, I overlooked that var HIDAPI was used in the submodule itself... I wonder now what my OSX and Linux tests actually did...

**Comment by brianlheim** [comment]
Thank you so much for this! I'd like to try building this myself over the weekend.

**Comment by bagong** [comment]
Thank you, @brianlheim , feedback much appreciated, of course. And don't miss out on the symlink PR in your test. With it no separation between sclang and C(++)-development is necessary...

File: .gitignore [hunk 1]
```diff
@@ -9,6 +9,7 @@ platform/mac/build
 # Windows related files
 *Thumbs.db
 CMakeLists.txt.user
+external_libraries/asio*
```
Comment (vivid-synth) [inline .gitignore]: Why are we ignoring these?

**Comment by vivid-synth** [comment]
@bagong am I right that you've also just fixed https://github.com/supercollider/supercollider/issues/2279 with this PR?

**Comment by bagong** [comment]
@vivid-synth - interesting point: on Windows I've made sure that the HID Windows-provided library is not linked if SC_HIDAPI is set to off. Honestly I am not sure what happened on the other OSSes if one used that option, it's likely not clean... I'll have a look during the weekend... Or if you'd like to share the effort... you can push to this PR in my repo.

File: .gitignore (hunk 1 above)
Comment (bagong) [inline .gitignore]: Thumbs.db is evident, right...
CMakeLists.txt.user is under Qt-Creator control, you wouldn't edit it.
asio* refers to the Steinberg ASIO-SDK - we use it in vanilla state, nobody would ever edit it (possibly even illegal), but it has to be in the source tree to be detected by portaudio.

I don't like "untracked files" that will never be added in my git status, and there is always the risk that somebody adds and commits them by accident (it has happened ;) )

**Comment by bagong** [comment]
So @vivid-synth , I think SC_HIDAPI now cleanly removes HID support from the build on Windows, Linux and OSX (Linux and OSX tested). I feel a bit bad because it was not my original intention to create this option, but I slid into it because of a build breakage in Windows. I had not intention to express that HIDAPI might be unnecessary ;)
I set the variable to advanced.

I know there are many open PRs. This one is on a level hardly interfering with SC innards, and in a domain not many developers are interested in. Oth committing this quickly will make other work in the pipeline a lot less complicated to set up, so may I request this to be committed quickly, once all observed problems are solved? Thank you!!

**Comment by brianlheim** [comment]
OK, I think this is ready to merge pending additional edits to the README. Here's what I've changed:

- Fix a warning message in cmake that warned about gcc-version < 5.3 even when the version was >= 5.3
- Overhaul and reorganize the Windows readme

I've tested and confirmed the 4-PR combo, including the SC_HIDAPI option, for these environments:

- Linux Fedora
- Windows (minGW, MSYS2, VS 2013)
- macOS Sierra

In the readme, I tried to be concise without being terse. I moved many conditional statements and optional settings to their own section or, where that wasn't possible, to flag them either through formatting or explicit statement that they're less important or "fine print". I also put the "troubleshooting" section above the "walkthroughs" section. I didn't touch much in the content of either of those parts. I updated some other small things I noticed were not quite right while working off the Readme this weekend. @bagong you were a huge help in editing this and of course if you think I've wrongly eliminated or misworded anything I'd love your feedback. :)

**Comment by brianlheim** [comment]
BTW, the TravisCI build failing for the push config is because I made a stupid mistake while pushing; the PR config should be taken as confirmation.

**Comment by bagong** [comment]
Thanks a lot @brianlheim , both for testing and adding enhancements!!

**Comment by brianlheim** [comment]
@bagong it was a pleasure, and I learned a lot in the process!

**Comment by vivid-synth** [comment]
@bagong is there a reason to feel bad for adding the HIDAPI option? Seems useful to me...

@brianlheim testing on 3 platforms? Dammmn! Thanks.

**Comment by bagong** [comment]
@vivid-synth , it's just an effort to be friendly to the authors of the new hidapi version, who spent a lot of time adding it to SC ;)

**Comment by brianlheim** [comment]
OK. After @bagong's review I am comfortable merging all 4 of these. Bon voyage, MSYS2!

**Pull request merged**